algosdk/atomic_transaction_composer.py

```diff
@@ -517,20 +582,33 @@ def execute(
         self.status = AtomicTransactionComposerStatus.COMMITTED
 
         confirmed_round = resp["confirmed-round"]
-        method_results = []
 
-        for i, tx_id in enumerate(self.tx_ids):
+        tx_results = [
+            client.pending_transaction_info(tx_id) for tx_id in self.tx_ids
```

Comment:
caveat/nit: I don't think we need to get the txn info for every transaction, just the ones with method calls (`self.method_dict`). If the call overhead isn't too bad, we could keep as written in the current PR.